frontend/packages/console-dynamic-plugin-sdk/src/extensions/dashboards.ts

```diff
@@ -0,0 +1,383 @@
+  e.type === 'console.dashboards/tab';
+
+export const isDashboardsCard = (e: Extension): e is ResolvedDashboardsCard =>
+  e.type === 'console.dashboards/card';
+
+export const isDashboardsOverviewHealthPrometheusSubsystem = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewHealthPrometheusSubsystem =>
+  e.type === 'console.dashboards/overview/health/prometheus';
+
+export const isDashboardsOverviewURLSubsystem = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewURLSubsystem =>
+  e.type === 'console.dashboards/overview/health/url';
+
+export const isDashboardsOverviewHealthResourceSubsystem = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewHealthResourceSubsystem =>
+  e.type === 'console.dashboards/overview/health/resource';
+
+export const isDashboardsOverviewHealthOperator = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewHealthOperator =>
+  e.type === 'console.dashboards/overview/health/operator';
+
+export const isDashboardsInventoryItemGroup = (
+  e: Extension,
+): e is ResolvedDashboardsInventoryItemGroup =>
+  e.type === 'console.dashboards/inventory/item/group';
+
+export const isDashboardsOverviewInventoryItem = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewInventoryItem => e.type === 'console.dashboards/inventory/item';
+
+export const isDashboardsOverviewResourceActivity = (
+  e: Extension,
+): e is ResolvedDashboardsOverviewResourceActivity =>
+  e.type === 'console.dashboards/overview/activity/resource';
+
+export type DashboardCardSpan = 4 | 6 | 12;
+
+export type GetOperatorsWithStatuses<R extends K8sResourceCommon = K8sResourceCommon> = (
+  resources: FirehoseResourcesResult,
+) => OperatorStatusWithResources<R>[];
+
+export type K8sActivityProps = {
+  resource: K8sResourceKind;
+};
+
+export type OperatorRowProps<R extends K8sResourceCommon = K8sResourceCommon> = {
+  operatorStatus: OperatorStatusWithResources<R>;
+};
+
+export type OperatorStatusWithResources<R extends K8sResourceCommon = K8sResourceCommon> = {
+  operators: R[];
+  status: OperatorStatusPriority;
+};
+
+export type OperatorStatusPriority = {
+  title: string;
+  priority: number;
+  icon: React.ReactNode;
+  health: keyof typeof HealthState;
+};
+
+export type PrometheusHealthHandler = (
+  responses: { response: PrometheusResponse; error: any }[],
+  t?: TFunction,
+  additionalResource?: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
+) => SubsystemHealth;
+
+export type PrometheusHealthPopupProps = {
+  responses: { response: PrometheusResponse; error: any }[];
+  k8sResult?: FirehoseResult<K8sResourceKind | K8sResourceKind[]>;
+};
+
+export type ResourceHealthHandler<R extends ResourcesObject> = (
+  resourcesResult: WatchK8sResults<R>,
+  t?: TFunction,
+) => SubsystemHealth;
+
+export type SubsystemHealth = {
+  message?: string;
+  state: HealthState;
+};
+
+export type URLHealthHandler<R> = (
+  response: R,
+  error: any,
+  additionalResource?: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
```

Comment:
will need a second generic for `additionalResource`


```suggestion
export type URLHealthHandler<R, T extends K8sResourceCommon | K8sResourceCommon[]> = (
  response: R,
  error: any,
  additionalResource?: FirehoseResult<T>,
```